xcode/Subconscious/Shared/Components/Common/Forms/ValidatedTextField.swift

```diff
@@ -54,16 +51,22 @@ struct ValidatedFormField<T: Equatable>: View {
                             .background(backgroundColor)
                     }
                     .padding(.trailing, 1)
-                    .opacity(isValid ? 0 : 1)
-                    .animation(.default, value: isValid)
+                    .opacity(field.shouldPresentAsInvalid ? 1 : 0)
+                    .animation(.default, value: field.shouldPresentAsInvalid)
                 }
                 .onChange(of: focused) { focused in
                     send(.focusChange(focused: focused))
+                    onFocusChanged(focused)
                 }
-                .onChange(of: innerText, perform: { innerText in
+                .onChange(of: innerText) { innerText in
                     send(.setValue(input: innerText))
-                })
-                .submitLabel(submitLabel)
+                }
+                .onChange(of: field) { field in
+                    // The has been reset, sync inner value
+                    if !field.touched && innerText != field.value {
+                        innerText = field.value
+                    }
```

Comment:
This is also the cause of the QR code scanning issue, attach it

Reply:
Approach LGTM. Would playtest a bit to make sure no infinite loops are created, but I've had to do two-way sync code like this before in SwiftUI to work around similar issues, and this looks correct.

Reply:
I didn't get it right the first time haha, my first _few_ attempts resulted in infinite loops 😅

This seems robust from my testing so far, but we'll see...